src/lib.rs

```diff
@@ -68,9 +67,25 @@ pub async fn run_trin(
         prometheus_exporter::start(addr)?;
     }
 
+    // Initialize validation oracle
+    let header_oracle = HeaderOracle::default();
+    info!(hash_tree_root = %hex_encode(header_oracle.header_validator.pre_merge_acc.tree_hash_root().0),"Loaded
+        pre-merge accumulator.");
+    let header_oracle = Arc::new(RwLock::new(header_oracle));
+
     // Initialize and spawn uTP socket
     let (utp_talk_reqs_tx, utp_talk_reqs_rx) = mpsc::unbounded_channel();
-    let discv5_utp_socket = Discv5UdpSocket::new(Arc::clone(&discovery), utp_talk_reqs_rx);
+
+    // Set the enr_cache_capacity to half the maximum uTP limit between all active networks. This is
+    // a trade off between memory usage and increased searches from the networks for each Enr.
+    let enr_cache_capacity =
+        portalnet_config.utp_transfer_limit * 2 * trin_config.portal_subnetworks.len() / 2;
```

Comment:
`portalnet_config.utp_transfer_limit * trin_config.portal_subnetworks.len()` this would be half.

as utp_transfer_limit is 2x for income and outbound limit